pkg/build/strategies/sti/sti.go

```diff
@@ -306,9 +306,13 @@ func (builder *STI) PostExecute(containerID, location string) error {
 }
 
 func (builder *STI) getPreviousImage() string {
+	if !builder.incremental || !builder.config.RemovePreviousImage {
+		return ""
+	}
```

Comment:
In case of the incremental build && when we need to remove previous image [we are saving the metadata at the beginning of the method and trying to remove image at the end of the method](https://github.com/openshift/source-to-image/blob/619e8798fc33a5f684201c716a12c41e69e74550/pkg/build/strategies/sti/sti.go#L276-L302). Also `previousImageId` can be empty if we couldn't fetch metadata for the image.

So, as far I understand, we can only move this `if` outside of the method. But I wouldn't like to do so, because I was trying to make `PostExecute()` more clean and contain only high-level methods with human-readable names.